frontend/src/components/ShootWarnings/ShootWarnings.vue

```diff
@@ -0,0 +1,327 @@
+          <template v-slot:activator="{ on }">
+            <v-icon v-on="on" :color="overallStatus" :small="small">{{icon}}</v-icon>
+          </template>
+          <span>{{tooltip}}</span>
+        </v-tooltip>
+      </v-btn>
+    </template>
+    <v-list>
+      <v-list-item
+        v-for="({key, icon, color, messageComponent, messageComponentProperties}) in warningItems"
+        :key="key">
+        <v-list-item-icon>
+          <v-icon :color="color">{{icon}}</v-icon>
+        </v-list-item-icon>
+        <v-list-item-content>
+          <component :is="messageComponent" v-bind="messageComponentProperties" class="message" />
+        </v-list-item-content>
+      </v-list-item>
+    </v-list>
+  </g-popper>
+</template>
+
+<script>
+
+import GPopper from '@/components/GPopper'
+import K8sExpirationMessage from '@/components/ShootWarnings/K8sExpirationMessage'
+import WorkerGroupExpirationMessage from '@/components/ShootWarnings/WorkerGroupExpirationMessage'
+import NoHibernationScheduleMessage from '@/components/ShootWarnings/NoHibernationScheduleMessage'
+import ClusterExpirationMessage from '@/components/ShootWarnings/ClusterExpirationMessage'
+import ConstraintWarningMessage from '@/components/ShootWarnings/ConstraintWarningMessage'
+import some from 'lodash/some'
+import get from 'lodash/get'
+import map from 'lodash/map'
+import forEach from 'lodash/forEach'
+import { shootItem } from '@/mixins/shootItem'
+import {
+  k8sVersionExpirationForShoot,
+  expiringWorkerGroupsForShoot,
+  isShootHasNoHibernationScheduleWarning,
+  isSelfTerminationWarning
+} from '@/utils'
+import { mapGetters } from 'vuex'
+
+export default {
+  name: 'ShootWarnings',
+  components: {
+    GPopper,
+    K8sExpirationMessage,
+    WorkerGroupExpirationMessage,
+    NoHibernationScheduleMessage,
+    ClusterExpirationMessage,
+    ConstraintWarningMessage
+  },
+  props: {
+    shootItem: {
+      type: Object
+    },
+    small: {
+      type: Boolean,
+      default: false
+    },
+    allWarnings: {
+      type: Boolean,
+      default: false
+    },
+    k8sWarning: {
+      type: Boolean,
+      default: false
+    },
+    machineImageWarning: {
+      type: Boolean,
+      default: false
+    },
+    noHibernationScheduleWarning: {
+      type: Boolean,
+      default: false
+    },
+    clusterExpirationWarning: {
+      type: Boolean,
+      default: false
+    },
+    hibernationConstraintWarning: {
+      type: Boolean,
+      default: false
+    },
+    maintenanceConstraintWarning: {
+      type: Boolean,
+      default: false
+    }
+  },
+  mixins: [shootItem],
+  computed: {
+    ...mapGetters([
+      'canPatchShoots'
+    ]),
+    visible () {
+      return !this.isShootMarkedForDeletion && (
+        this.isK8sWarning ||
+        this.isMachineImageWarning ||
+        this.isNoHibernationScheduleWarning ||
+        this.isClusterExpirationWarning ||
+        this.isConstraintWarning
+      )
+    },
+    warningItems () {
```

Comment:
What is the difference between a warning which is a warning and a warning which is an error or and just an info. That's why I named it messages. It is unclear to me what we call message and what we call warning and how an error could be a warning.

Reply:
done, do not use term _warning_ for component and generic methods as this is misleading. Use term _message_